src/profile-logic/processed-profile-versioning.js

```diff
@@ -781,5 +781,40 @@ const _upgraders = {
       thread.markers.data = newDataArray;
     }
   },
+  [17]: profile => {
+    // funcTable gains a new field: columnNumber.
+    for (const thread of profile.threads) {
+      const { funcTable, stringArray } = thread;
+      const stringTable = new UniqueStringArray(stringArray);
+
+      funcTable.columnNumber = [];
+      for (
+        let funcIndex = 0;
+        funcIndex < thread.funcTable.length;
+        funcIndex++
+      ) {
+        if (funcTable.isJS[funcIndex]) {
+          const fileNameIndex = funcTable.fileName[funcIndex];
+          let fileName;
+          if (fileNameIndex !== null) {
+            fileName = stringTable.getString(fileNameIndex);
+            const match = /^(.*):([0-9]+)$/.exec(fileName);
+            if (match) {
+              funcTable.columnNumber[funcIndex] =
+                funcTable.lineNumber[funcIndex];
+              stringTable.changeString(fileNameIndex, match[1]);
```

Comment:
as said below, we should use `indexForString` for the new string, and use the new index, instead of mutating the array. We _should_ get often the same index because we _should_ have often the same string :-)